pkg/stanza/fileconsumer/internal/archive/archive.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package archive // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/archive"
+
+import (
+	"os"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fingerprint"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+)
+
+const _ = "knownFiles"
+
+type Archive interface {
+	SetStorageClient(operator.Persister)
+
+	Match([]*FileRecord) ([]*FileRecord, error)
+	Write([]*reader.Metadata) error
+}
+
+type FileRecord struct {
+	File        *os.File
+	Fingerprint *fingerprint.Fingerprint
```

Comment:
@djaglowski storing file handle is crucial for us because of the way we handle the archiving:

1. We try to match files against `knownFiles` array.
2. If a match isn't found, we would accumulate the unmatched files (instead of finding in the archive. More about it was discussed [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32727#issuecomment-2260891596).
3. Perform the archiving logic. Described [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/431ab4163432fe03658ddd0f34e89598db4abad7/pkg/stanza/fileconsumer/internal/archive/archive.go#L51-L59).

We need the file handle so that we can create a reader from it after the archiving is done.

Reply:
This doesn't make any sense to me. See my comments on the other PR. I think this is a fundamental point that needs to be resolved before bothering with the rest of the PR.

`reader.Metadata` is sufficient and necessary for creating readers in all other circumstances _except where we already have an open file handle_, in which case we can do better. 

I do not understand why we would or even could have open file handles at this point, since the archive should only be populated by elements which are discarded from `knownFiles`, which only contains `reader.Metadata`.